### PR TITLE
Visualize Graphs After Each Pass

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: src/ngraph/pass/manager.cpp [hunk 1]
```diff
@@ -89,6 +96,16 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
                 call_graph_pass->run_on_call_graph(f->get_ordered_ops());
             }
         }
+
+        if (visualize)
+        {
+            //visualizations will be named after the outermost function
+            auto fname = fs.at(0)->get_name() + std::string("_") + m_pass_names.at(index) +
```
Comment (rkimballn1) [inline src/ngraph/pass/manager.cpp]: pass_names...genius!

File: src/ngraph/pass/manager.hpp [hunk 2]
```diff
@@ -48,13 +49,19 @@ class ngraph::pass::Manager
...
 
     void run_passes(std::shared_ptr<Function>);
 
     ManagerState& get_state();
 
 private:
+    std::vector<std::string> m_pass_names;
     std::vector<std::shared_ptr<PassBase>> m_pass_list;
     ManagerState m_state;
+    bool visualize = false;
```
Comment (rkimballn1) [inline src/ngraph/pass/manager.hpp]: Come on! You know the naming conventions. `m_visualize`

File: src/ngraph/pass/manager.cpp [hunk 3]
```diff
@@ -25,12 +25,18 @@
...
 #include "ngraph/pass/manager.hpp"
 #include "ngraph/pass/pass.hpp"
+#include "ngraph/pass/visualize_tree.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 ngraph::pass::Manager::Manager()
 {
+    static const auto nevt = std::getenv("NGRAPH_ENABLE_VISUALIZE_TRACING");
```
Comment (rkimballn1) [inline src/ngraph/pass/manager.cpp]: Would be nice to have a programmatic way to set this too.

Comment (Krovatkin) [reply src/ngraph/pass/manager.cpp]: add a setter so visualization can be turned on and off

**Pull request merged**